llvm/lib/CodeGen/RegisterCoalescer.cpp

```diff
@@ -456,6 +456,30 @@ static bool isSplitEdge(const MachineBasicBlock *MBB) {
   return true;
 }
 
+static const TargetRegisterClass *
+getLargestLegalRegClass(Register Reg, const MachineFunction *MF,
+                        const MachineRegisterInfo &MRI) {
```

Comment:
This is mostly copy paste from the MRI function, can you refactor it there to expose this